examples/wasi-http-rust-async/src/lib.rs

```diff
@@ -0,0 +1,290 @@
+wit_bindgen::generate!("proxy" in "../../wit/preview2");
```

Comment:
I think we can definitely add async/await support to JS and Python.  I'm not sure how it would work with TinyGo, though, since I have limited experience with Go.  My understanding is that the most natural way to do concurrent I/O in Go is via goroutines, but I don't know how that's supported in TinyGo when targeting WASI or how we would hook into it with a `poll_oneoff`-based solution.  Do you know anything about that?

Reply:
I don't. I'd like to find it out myself.